renku/api/datasets.py

```diff
@@ -79,59 +91,66 @@ def datasets(self):
             result[path] = Dataset.from_yaml(path)
         return result
 
-    @contextmanager
-    def with_dataset(self, name=None):
-        """Yield an editable metadata object for a dataset."""
+    def load_dataset(self, name=None):
+        """Load dataset reference file."""
         from renku.models.refs import LinkReference
+        path = None
+        dataset = None
 
-        with self.lock:
-            path = None
-            dataset = None
+        if name:
+            path = self.renku_datasets_path / name / self.METADATA
 
-            if name:
-                path = self.renku_datasets_path / name / self.METADATA
+            if not path.exists():
+                path = LinkReference(
+                    client=self, name='datasets/' + name
+                ).reference
 
-                if not path.exists():
-                    path = LinkReference(
-                        client=self, name='datasets/' + name
-                    ).reference
+            if path.exists():
+                with path.open('r') as f:
+                    source = yaml.safe_load(f) or {}
+                dataset = Dataset.from_jsonld(source, __reference__=path)
 
-                if path.exists():
-                    with path.open('r') as f:
-                        source = yaml.safe_load(f) or {}
-                    dataset = Dataset.from_jsonld(source, __reference__=path)
+        if dataset is None:
```

Comment:
maybe this part should still be part of `with_dataset` to avoid side effects and the function should raise NotFound or return None when there is no dataset.

Reply:
@jsam why not moving it to `with_dataset` method line 143? The `load_dataset` should raise or return None, but `with_dataset` is expected to create a dataset (which are the lines 114:127).